src/main/java/org/spongepowered/api/command/args/CommandFlags.java

```diff
@@ -486,4 +490,41 @@ public CommandElement buildWith(CommandElement wrapped) {
                     this.unknownLongFlagBehavior, this.anchorFlags);
         }
     }
+
+    private static class FlagElement extends CommandElement {
```

Comment:
Unfortunately it can't as is - I thought that too when I started, but after trying it I realized that the flag being parsed is not necessarily the key used for that flag either. For example, in the flags `xzvf` the key used is `x`. The flag `-v` should then be recognized as the same as flag `x`, which is what's added to the context in the same way as elements.

At the `CommandFlags` level, we don't really have this information anymore - just a map of flags to elements. The element should contain the correct key for non-value flags, but for value flags it could be anything, hence this PR and why it's can't be done this way.

One alternative here is to also send `CommandFlags` a map of flags to keys, which would allow us to look up the key. This seemed a bit too costly so I went with this approach. As far as the API goes, this wouldn't be visible as well (and it replaces the mark true element used previously).

Reply:
You could do it at this point and remove the addition of the key from everywhere else. 

https://github.com/SpongePowered/SpongeAPI/blob/9067212f054eee4187c623cca520f1d8c063f3f2/src/main/java/org/spongepowered/api/command/args/CommandFlags.java#L83

However, if a command has specified x and v as the same, then why do they need to know that x and v have both been specified? Surely thy only need to know the intended outcome?

The more I think about this, the more I think this really isn't needed. It's not as if a command doesn't have full control over the parameters it requires. With each flag, a value is placed into the context, which will be placed against a key that the developer should be aware of, and if they duplicate the key in other elements, thats on them.

I just don't see the use case. If you need to differentiate between -x and -v, then make them two separate flags, you should not be relying on the parser to tell you that separately.

Reply:
So at the point of `remove` we don't have the actual flags anyways - additional work would have to be done to re-extract them. Regardless, the main problem is that we can't use the flag the user entered for this when working with flag aliases. If you added the user input to the context, then `-x` and `-v` would be separate when they should be aliases.

The additional work done with the `FlagsElement` is so aliases of a flag still add the same key to the context. It's essentially what `markTrue` did with an additional step. This is what ensures the context *doesn't* differentiate between `-x` and `-v` because they're aliases for the same flag.

Reply:
Ok, so you want the first alias to be added to the flag set. I would still prefer this being information entirely within the command flags class.

Before you do though, the question still remains, is there a use case? I do not see one that should be solved with careful command planning. What does having a “hasFlag” method solve that you can’t solve with flags and their associated values in the command context?

Reply:
If it's done in `CommandFlags`, we need to also add a map for looking up the key as well, performing a lookup twice (or alternatively store a tuple instead).

This is useful when you have two flags which share the same value but are used differently. An example of this is defining a range for a random number command.

```
/rand {--lowerInclusive=<lower>| --lowerExclusive=<lower> | --upperInclusive=<upper> | --upperExclusive=<lower>}

/rand --li 1 --ui 9
/rand --li 0 --ue 10
```
The handling for this would be:
```
int lower = args.requireOne("lower");
int upper = args.requireOne("upper");
BoundType lBound = args.hasFlag("lowerInclusive") ? BoundType.CLOSED : BoundType.OPEN;
BoundType uBound = args.hasFlag("upperInclusive") ? BoundType.CLOSED : BoundType.OPEN;
Range range = Range.range(lower, lBound, upper, uBound);
```
It's a bit contrived of course, but I think for this particular example it's the cleanest way to express this command.

Also worth noting is having `args.hasFlag("key")` rather than `args.hasAny("key")`, which might be cleaner from an API perspective.